pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -436,49 +468,99 @@ func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, p
 
 		// Something has changed, so track the new state and perform a sync.
 		glog.V(2).Infof("syncing resource quota controller with updated resources from discovery: %v", newResources)
-		oldResources = newResources
 
 		// Ensure workers are paused to avoid processing events before informers
 		// have resynced.
 		rq.workerLock.Lock()
 		defer rq.workerLock.Unlock()
 
+		// Resetting the REST mapper will also invalidate the underlying discovery
+		// client. This is a leaky abstraction and assumes behavior about the REST
+		// mapper, but we'll deal with it for now.
+		rq.restMapper.Reset()
```

Comment:
I thought we passed a restmapper that was already being periodically refreshed.  Why do we need to punch this out?

Reply:
I think we do, outside of the consuming controllers.

Reply:
> already being periodically refreshed

The `Sync` is supposed to periodically refresh the restmapper with new discovery information.

The reset occurs within this `Sync` function.